py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/batcher.py

```diff
@@ -77,6 +100,70 @@ def event_size_over_limit(self) -> bool:
         return False
 
 
+def compress_filenames(filenames: List[str]) -> Tuple[str, str, List[str]]:
+    """
+    compress a list of filenames into a dict with the prefix, suffix, and list
+    of unique components
+    """
+    if len(filenames) == 0:
+        return ("", "", [])
+
+    def longest_common(filenames: List[str], prefix: bool = True) -> str:
```

Comment:
function to get the longest common beginning or ending string of all files in the list.